src/lib/components/AccessRightField.js

```diff
@@ -139,7 +140,7 @@ FormikAccessRightField.defaultProps = {
 
 
 const mapStateToProps = (state) => ({
-  isMetadataOnly: !state.files.enabled,
+  isMetadataOnly: _isEmpty(state.files.entries),
```

Comment:
I thought so too before, but that's not the UX we want. We want that `isMetadataOnly` *mean* it is metadata-only i.e. the checkbox for metadata-only has been checked: https://github.com/inveniosoftware/invenio-app-rdm/issues/740 . Files protection should still be shown if no files but not metadata-only.

I think this should do it:

```suggestion
  isMetadataOnly: !state.deposit.record.files.enabled,
```

Reply:
It makes sense, your suggestion works flawless, thanks!